src/wire/mod.rs

```diff
@@ -95,6 +95,8 @@ mod ipv6fragment;
 mod icmpv4;
 #[cfg(feature = "proto-ipv6")]
 mod icmpv6;
+#[cfg(all(feature = "proto-ipv4", feature = "proto-ipv6"))]
```

Comment:
Rely on the `icmpv6` module for `proto-ipv6`.

Reply:
The comment by @whitequark [here](https://github.com/m-labs/smoltcp/pull/205/files#r187041666) was kind of what I was getting at and is a much clearer. If we get an ICMPv6 response we shouldn't be using the definitions in `wire::icmpv4`.

Note: my assumption was that `wire::icmp` was the same thing as `wire::icmpv4`. If your intent was to make `wire::icmp` a module that encapsulates ICMPv4 and ICMPv4 (similar to `wire::ip` for IPv4 and IPv6), then your current approach is correct.

Reply:
Hmm, except that it should be `any()` and not `all()`

Reply:
True.